clouddq/runners/dbt/dbt_runner.py

```diff
@@ -123,93 +117,79 @@ def get_entity_summary_path(self) -> Path:
+            "environment_target": self.environment_target
+        }
 
-    def get_dq_summary_dataset_region(self) -> str:
+    def get_dq_summary_dataset_region(self, gcp_project_id, gcp_bq_dataset_id ) -> str:
         self._resolve_connection_configs(
-            dbt_profiles_dir=self.dbt_profiles_dir,
+            gcp_project_id=gcp_project_id,
+            gcp_bq_dataset_id=gcp_bq_dataset_id,
             environment_target=self.environment_target,
         )
         Path(self.dbt_profiles_dir)
         return self.environment_target
 
     def _resolve_connection_configs(
         self,
-        dbt_profiles_dir: Optional[str],
+        gcp_project_id: str,
+        gcp_bq_dataset_id: str,
         environment_target: Optional[str],
         bigquery_client: Optional[BigQueryClient] = None,
-        gcp_project_id: Optional[str] = None,
         gcp_region_id: Optional[str] = None,
-        gcp_bq_dataset_id: Optional[str] = None,
         gcp_service_account_key_path: Optional[Path] = None,
         gcp_impersonation_credentials: Optional[str] = None,
     ) -> None:
-        if dbt_profiles_dir:
-            dbt_profiles_dir = Path(dbt_profiles_dir).absolute()
-            if not dbt_profiles_dir.joinpath("profiles.yml").is_file():
-                raise ValueError(
-                    f"Cannot find connection `profiles.yml` configurations at "
-                    f"`dbt_profiles_dir` path: {dbt_profiles_dir}"
-                )
-            self.dbt_profiles_dir = dbt_profiles_dir
+        # create GcpDbtConnectionConfig
+        connection_config = GcpDbtConnectionConfig(
+            gcp_project_id=gcp_project_id,
+            gcp_bq_dataset_id=gcp_bq_dataset_id,
+            bigquery_client=bigquery_client,
+            gcp_region_id=gcp_region_id,
+            gcp_service_account_key_path=gcp_service_account_key_path,
+            gcp_impersonation_credentials=gcp_impersonation_credentials,
+        )
+        self.connection_config = connection_config
+        self.dbt_profiles_dir = Path(self.dbt_path)
+        logger.debug(
+            "Using dbt profiles.yml path: {self.dbt_profiles_dir}",
+        )
+        if environment_target:
+            logger.debug(f"Using `environment_target`: {environment_target}")
             self.environment_target = environment_target
-        else:
-            # create GcpDbtConnectionConfig
-            connection_config = GcpDbtConnectionConfig(
-                gcp_project_id=gcp_project_id,
-                gcp_bq_dataset_id=gcp_bq_dataset_id,
-                bigquery_client=bigquery_client,
-                gcp_region_id=gcp_region_id,
-                gcp_service_account_key_path=gcp_service_account_key_path,
-                gcp_impersonation_credentials=gcp_impersonation_credentials,
+            self.connection_config.to_dbt_profiles_yml(
+                target_directory=self.dbt_profiles_dir,
+                environment_target=self.environment_target,
             )
```

Comment:
this can be moved outside of the if else and called only once

Reply:
updated.